### PR TITLE
cpp-coverals will now raise an exception if neither the travis job id nor the repo token are found

### PR DESCRIPTION
**Pull request by mkfifo**
This addresses the initial work raised in issue #82 but in a way that works with the existing travis workflow (travis job id)

### CODE REVIEW

File: cpp_coveralls/__init__.py [hunk 1]
```diff
@@ -84,7 +84,13 @@ def run():
 
     if not args.no_gcov:
         coverage.run_gcov(args)
+
     cov_report = coverage.collect(args)
+
+    if not cov_report.get("repo_token") and not cov_report.get("service_job_id"):
```
Comment (eddyxu) [inline cpp_coveralls/__init__.py]: Can you raise the `Error` after parsing argument? We don't need to run `gcov` before finding out one of these fields is missing.

**Comment by mkfifo** [comment]
@eddyxu good point, thanks for that.

I have made a new push that should raise the error in the correct location (after arg parsing and before any gcov).

**Comment by mkfifo** [comment]
re-pushed, had a silly mistake in my commit.

File: cpp_coveralls/__init__.py [hunk 2]
```diff
@@ -82,9 +82,15 @@ def run():
 
     args.service_job_id = os.environ.get('TRAVIS_JOB_ID', '')
 
+    if args.repo_token == '' and args.service_job_id == '':
+        raise ValueError("\nno coveralls.io token specified and no travis job id found\n"
+                         "see --help for examples on how to specify a token\n")
+
     if not args.no_gcov:
         coverage.run_gcov(args)
+
```
Comment (eddyxu) [inline cpp_coveralls/__init__.py]: The commit looks very good. Just a few minor comments:

* Could you remove the blank line changes here.
* Could you rebase to merge you changes into one commit?

File: test.bash [hunk 3]
```diff
@@ -24,6 +24,7 @@ int main() {
 EOF
 gcc -coverage -o foo foo.c
 ./foo
+export COVERALLS_REPO_TOKEN="fake testing token"
```
Comment (eddyxu) [inline test.bash]: Could you test whether the 'TRAVIS_JOB_ID` is set before set repo token?

We also want to test this logic on a travis machine.

**Comment by mkfifo** [comment]
@eddyxu sorry about that, I was meaning to squash them down once I had all the kinks worked out.

I have pushed a new squashed fix, I have also updated the test script to work in both travis and non-travis environments, further I moved the COVERALLS_REPO_TOKEN handling code to be in __init__.py:run along with the yaml handling code (as it was in coverage.py:collect which is called too late).

I removed the whitespace as this commit is meant to be functional only, however stylistically I do dislike how tightly packed all the branching code in __init__.py:run is, especially considering the lack of comments as it makes for a very hard read.

**Comment by eddyxu** [comment]
Cool, @mkfifo  Thanks a lot for working on this!

**Pull request merged**